### PR TITLE
cypher: Add cypher output using neo4j driver

### PR DESCRIPTION
**Pull request by rockwotj**
Fixes: https://github.com/redpanda-data/connect/issues/869

### CODE REVIEW

**Comment by rockwotj** [comment]
I am going to add an integration test first.

**Comment by rockwotj** [comment]
Added an integration test 🎉

File: internal/plugins/info.csv [hunk 1]
```diff
@@ -56,6 +56,7 @@ couchbase               ,cache     ,Couchbase               ,4.12.0  ,community
 couchbase               ,processor ,Couchbase               ,4.11.0  ,community  ,n          ,n     ,n
 csv                     ,input     ,csv                     ,0.0.0   ,certified  ,n          ,n     ,n
 csv                     ,scanner   ,csv                     ,0.0.0   ,certified  ,n          ,y     ,y
+cypher                  ,output    ,cypher                  ,0.0.0   ,community  ,n          ,n     ,n
```
Comment (Jeffail) [inline internal/plugins/info.csv:59]: We can probably safely set the version to `4.37.0`. I'm not sure what our process is going to look like for determining when to enable for cloud, maybe worth @vladoschreiner chiming in during the PR phase.

**Review by mihaitodor — APPROVED** [review]
Thanks @rockwotj! I left a few small comments, but it should be good to go otherwise.

File: internal/impl/cypher/output.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+	var argsMapper *service.MessageBatchBloblangExecutor
+	if o.argsMapping != nil {
+		argsMapper = batch.BloblangExecutor(o.argsMapping)
+	}
+	_, err := session.ExecuteWrite(ctx, func(tx neo4j.ManagedTransaction) (any, error) {
+		for i, msg := range batch {
+			mapped := msg
+			if argsMapper != nil {
+				var err error
+				mapped, err = argsMapper.Query(i)
```
[the remark this answers is not in the document]
Comment (rockwotj) [reply internal/impl/cypher/output.go:250]: I think I did that here right? On line 243?

File: internal/impl/cypher/output.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+	}
+	if err := driver.VerifyAuthentication(ctx, nil); err != nil {
+		return fmt.Errorf("unable to verify correct authentication: %w", err)
+	}
+	o.driver = driver
+	return nil
+}
+
+func (o *output) WriteBatch(ctx context.Context, batch service.MessageBatch) error {
+	session := o.driver.NewSession(ctx, neo4j.SessionConfig{
```
[the remark this answers is not in the document]
Comment (rockwotj) [reply internal/impl/cypher/output.go:235]: I peeked at the underlying implementation and I believe it's all safe under the hood.

File: internal/impl/cypher/output.go [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+			if argsMapper != nil {
+				var err error
+				mapped, err = argsMapper.Query(i)
+				if err != nil {
+					return nil, fmt.Errorf("unable to execute %s: %w", coFieldArgsMapping, err)
+				}
+			}
+			data, err := mapped.AsStructured()
+			if err != nil {
+				return nil, err
```
[the remark this answers is not in the document]
Comment (rockwotj) [reply internal/impl/cypher/output.go:257]: Good call. Done

File: internal/impl/cypher/output.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mihaitodor) [reply internal/impl/cypher/output.go:250]: Derp... My bad! Need to get my glasses checked. Thanks Tyler!

File: internal/plugins/info.csv (hunk 1 above)
Comment (rockwotj) [reply internal/plugins/info.csv:59]: I pinged in slack, seems like the process is initially no until the security team reviews... 🤷

File: internal/impl/cypher/output.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mihaitodor) [reply internal/impl/cypher/output.go:235]: OK, thanks for checking!

File: internal/plugins/info.csv (hunk 1 above)
Comment (rockwotj) [reply internal/plugins/info.csv]: Set the version, thanks!

**Pull request merged**